### PR TITLE
Fix virt-customize resize partition failure

### LINKED ISSUE #133: guest-tools: image: virt-customize fails to create the guest image
```
sdp@984fee00bd70:~/tdx/guest-tools/image$ sudo -E ./create-td-image.sh 
SUCCESS: Installation of required tools
--2024-06-05 04:59:22--  https://cloud-images.ubuntu.com/releases/noble/release//SHA256SUMS
Resolving proxy-dmz.intel.com (proxy-dmz.intel.com)... 10.7.211.16
Connecting to proxy-dmz.intel.com (proxy-dmz.intel.com)|10.7.211.16|:912... connected.
Proxy request sent, awaiting response... 200 OK
Length: 7629 (7.5K)
Saving to: ‘SHA256SUMS’

SHA256SUMS                                            100%[======================================================================================================================>]   7.45K  --.-KB/s    in 0.01s   

2024-06-05 04:59:23 (506 KB/s) - ‘SHA256SUMS’ saved [7629/7629]

SUCCESS: Verify the checksum for Ubuntu cloud image.
SUCCESS: Copy the ubuntu-24.04-server-cloudimg-amd64.img => /tmp/tdx-guest-ubuntu-24.04-generic.qcow2
Image resized.
[   0.0] Examining the guest ...
virt-customize: error: libguestfs error: passt exited with status 1

If reporting bugs, run virt-customize with debugging enabled and include 
the complete output:

  virt-customize -v -x [...]
WARN: Failed to resize guest image to 50G
~/tdx/guest-tools/image/cloud-init-data ~/tdx/guest-tools/image
SUCCESS: Generate configuration for cloud-init...
I: -input-charset not specified, using utf-8 (detected in locale settings)
Total translation table size: 0
Total rockridge attributes bytes: 331
Total directory bytes: 0
Path table size(bytes): 10
Max brk space used 0
183 extents written (0 MB)
SUCCESS: Generate the cloud-init ISO image...
~/tdx/guest-tools/image
SUCCESS: Complete cloud-init...
[   0.0] Examining the guest ...
virt-customize: error: libguestfs error: passt exited with status 1

If reporting bugs, run virt-customize with debugging enabled and include 
the complete output:

  virt-customize -v -x [...]
ERROR: Failed to setup guest image
SUCCESS: Cleanup!
```

### PR DESCRIPTION
**Pull request by hector-cao**
Fixes #133

### CODE REVIEW

File: guest-tools/image/create-td-image.sh [hunk 1]
```diff
@@ -180,11 +180,30 @@ create_guest_image() {
     resize_guest_image
 }
 
+# To resize the guest image
+# 1) we add additionnal space to the qcow image using qemu-img tool
```
Comment (BFuhry) [inline guest-tools/image/create-td-image.sh:184]:
```suggestion
# 1) we add additional space to the qcow image using qemu-img tool
```

File: guest-tools/image/create-td-image.sh [hunk 2]
```diff
@@ -180,11 +180,30 @@ create_guest_image() {
...
+# 2) we extend (using growpart) the partition sda1 to fill empty space until end of disk
+#    since sda1 is the last partition, it will take all space we previously added
+# 3) we resize the file system to cover all partition space
+#
+# NB: We should not use static name for the disk device (sda) because it can
+# change on boot (the main disk is named sdb). Using sda naming can cause failure
```
Comment (BFuhry) [inline guest-tools/image/create-td-image.sh:190]: Do you mean: "e.g., the main disk might be named sdb"?

Comment (hector-cao) [reply guest-tools/image/create-td-image.sh:190]: Yes

Comment (BFuhry) [reply guest-tools/image/create-td-image.sh:190]:
```suggestion
# change on boot (e.g., the main disk might be named sdb). Using sda naming can cause failure
```
Then, I'd propose to be more explicit.

**Review by bktan8 — APPROVED** [review]
LGTM

**Pull request merged**